sunpy/util/scraper.py

```diff
@@ -90,10 +94,31 @@ def __init__(self, pattern, regex=False, **kwargs):
                 milli=milliseconds_,
                 end=self.pattern[milliseconds.end():]
             ))
+        self.extractor = extractor
+        self.matcher = matcher
 
     def matches(self, filepath, date):
         return date.strftime(self.pattern) == filepath
 
+    def infoextract(self, url, timestamp=False):
```

Comment:
If timestamp is `true`, we return a url dict containing time info. Else we return all fields except time.